File: platform-instances/authorization-server/src/main/java/com/sv/youapp/service/authorization/configuration/WebSecurityConfig.java
package com.sv.youapp.service.authorization.configuration;

import java.util.List;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.annotation.Order;
import org.springframework.security.config.Customizer;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.oauth2.server.authorization.config.annotation.web.configurers.OAuth2AuthorizationServerConfigurer;
import org.springframework.security.web.SecurityFilterChain;
import org.zalando.logbook.Logbook;

@Configuration
public class WebSecurityConfig {
  @Bean
  @Order(1)
  public SecurityFilterChain authorizationServerSecurityFilterChain(
      HttpSecurity http, List<AbstractHttpConfigurer<?, HttpSecurity>> configurers, Logbook logbook)
      throws Exception {
    OAuth2AuthorizationServerConfigurer authorizationServerConfigurer =
        OAuth2AuthorizationServerConfigurer.authorizationServer();

    http.securityMatcher(authorizationServerConfigurer.getEndpointsMatcher())
        .with(authorizationServerConfigurer, config -> {})
        .authorizeHttpRequests(auth -> auth.anyRequest().authenticated())
        .csrf(AbstractHttpConfigurer::disable)
        .formLogin(Customizer.withDefaults());
    for (AbstractHttpConfigurer<?, HttpSecurity> c : configurers) {
      http.with(c, config -> {});
    }
    return http.build();
  }

  @Bean
  @Order(2)
  public SecurityFilterChain defaultSecurityFilterChain(HttpSecurity http) throws Exception {
    http.authorizeHttpRequests(auth -> auth.anyRequest().permitAll())
        .csrf(AbstractHttpConfigurer::disable)
			.formLogin(formLogin -> formLogin.loginPage("/login").permitAll());
    return http.build();
  }
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/service/authorization/configuration/WebSecurityConfig.kt
package com.sv.youapp.service.authorization.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.Order
import org.springframework.security.config.Customizer
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer
import org.springframework.security.oauth2.server.authorization.config.annotation.web.configurers.OAuth2AuthorizationServerConfigurer
import org.springframework.security.web.SecurityFilterChain

@Configuration
class WebSecurityConfig {
    @Bean
    @Order(1)
    fun authorizationServerSecurityFilterChain(
        http: HttpSecurity,
        configurers: List<AbstractHttpConfigurer<*, HttpSecurity>>,
    ): SecurityFilterChain {
        val authorizationServerConfigurer: OAuth2AuthorizationServerConfigurer =
            OAuth2AuthorizationServerConfigurer.authorizationServer()
        http
            .securityMatcher(authorizationServerConfigurer.endpointsMatcher)
            .with(authorizationServerConfigurer) {}
            .authorizeHttpRequests { auth ->
                auth.anyRequest().authenticated()
            }.csrf { csrf -> csrf.disable() }
            .formLogin(Customizer.withDefaults())
        configurers.forEach {
            http.with(it) {}
        }
        return http.build()
    }

    @Bean
    @Order(2)
    fun defaultSecurityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http
            .authorizeHttpRequests { auth -> auth.anyRequest().authenticated() }
            .csrf { csrf -> csrf.disable() }
            .formLogin(Customizer.withDefaults())
        return http.build()
    }
}
